### PR TITLE
Fix some of the reported blocks not working

### LINKED ISSUE #2954: Rainbow blocks still work after disabling them
## :round_pushpin: Description (REQUIRED)
When disabling the rainbow blocks they still work. Work as in change colors
This has been tested with all SF rainbow blocks

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)
https://youtu.be/oryRBTfbh1o
In the video it shows the full steps to reproduce this issue

## :bulb: Expected behavior (REQUIRED)
to not change colors after toggling them off in console

## :scroll: Server Log
N/a

## :open_file_folder: /error-reports/ Folder
N/A

## :compass: Environment (REQUIRED)

image

### LINKED ISSUE #2957: Auto crafter still working after being disabled
## :round_pushpin: Description (REQUIRED)
So i was also testing the enhanced auto crafter. soo it works when disabled. what works
Set a recipe in the enhanced auto crafter
stop the server
disable the auto crafter
start the server
you can open the gui.
You can also toggle of and on stuff inside the gui. 
but when the the recipe is turned of the machine breaks after a certain amount of time.
but it does display the wrong message. when trying to use the disabled auto crafter.
for example right clicking with empty hand shows 
image
shift right clicking shows this message
image
but when you do it with an item that shouldnt work on it like a vanilla chest it shows this
image
Another update on this when the machine is disabled and you interact with it you can steal the items inside of it
i just obtained a barrier

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)
Thanks to walshy for uploading the vid blobs.are-pretty.sexy/7FbFgdR.mp4

## :bulb: Expected behavior (REQUIRED)

To not work after being disabled

## :scroll: Server Log
N/A

## :open_file_folder: /error-reports/ Folder
N/A

## :compass: Environment (REQUIRED)
image

### LINKED ISSUE #2958: Carbonado edge furnace still working after being disabled
## :round_pushpin: Description (REQUIRED)
When the carbonado edged furnace has been disabled with items still inside of it. It will still work with the speed the machine is normally it wont work on default furnace speed as you would expect.

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)
Place down carbonado edged furnace with items inside of it.
Stop server
disable the furnace
start server
Furnace is still smelting at the speed of carbonado edged but it shouldnt

## :bulb: Expected behavior (REQUIRED)
Furnace to act like a normal furnace or to just being disabled and canceling the smelting

## :scroll: Server Log
N/A

## :open_file_folder: /error-reports/ Folder
N/A

## :compass: Environment (REQUIRED)
image

### LINKED ISSUE #2959: Blockplacer still functions after being disabled
## :round_pushpin: Description (REQUIRED)
Blockplacer still functioning even when disabled if it has items inside of it

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)
Place blockplacer
place items inside of it
stop server
disable the blockplacer
start server
switch the lever to power the blockplacer
Sees blocks being placed

## :bulb: Expected behavior (REQUIRED)
Dispenser to stop working fully or to just work as a dispenser

## :scroll: Server Log
N/A

## :open_file_folder: /error-reports/ Folder
N/A

## :compass: Environment (REQUIRED)
image

### LINKED ISSUE #3011: Infused Hopper still works after being disabled.
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Infused Hopper still works after being disabled.

### 📑 Reproduction Steps

1. Place Infused Hopper
2. Stop the server
3. Disable it on items.yml
4. Start the server
5. See it's still sucking items

### 💡 Expected Behavior

It would not suck the items.

### 📷 Screenshots / Videos

https://youtu.be/XlHl3vb0T8s

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

image

### 🧭 Other plugins

_No response_

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

This PR should resolve all blocks not working if they are disalbed.
Continuation of #2968

## Proposed changes

Added canuse/isdisabledin checks on serveral items.

## Related Issues (if applicable)

Fixes #2957 
Fixes #2958 
Fixes #2959 
Fixes #3011
Fixes #2954 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by J3fftw1** [comment]
This PR does not fix #2954 just yet.
I cant completely figure that 1 out yet

I can not add #isDisabled on the slimefunitem.
Like @WalshyDev had done on https://github.com/Slimefun/Slimefun4/pull/2968/files#diff-4e447aa9b9b580f5d12d97b637552a38332628f43f120fb66cf50f4b7c18c91eR816
It throws a bunch of errors on start up and tests are also failing.

Any ideas on this?

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java [hunk 1]
```diff
@@ -536,6 +536,7 @@ private final void onEnable() {
     }
 
     private void loadItemHandlers() {
+
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java:539]: ?

**Comment by JustAHuman-xD** [comment]
> It throws a bunch of errors on start up and tests are also failing.

Where are you adding the isDisabled that causes the issues?

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java (hunk 1 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java:539]: whoops left overs from testing

**Comment by J3fftw1** [comment]
> > It throws a bunch of errors on start up and tests are also failing.
> 
> Where are you adding the isDisabled that causes the issues?



> > It throws a bunch of errors on start up and tests are also failing.
> 
> Where are you adding the isDisabled that causes the issues?

I tried to add it here.
https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java#L805

**Comment by JustAHuman-xD** [comment]
So it's because the item isn't registered there for it can't be disabled..... hmmmmm

**Comment by J3fftw1** [comment]
> So it's because the item isn't registered there for it can't be disabled..... hmmmmm

exactly

**Comment by JustAHuman-xD** [comment]
Why are you trying to add it to the addItemHandler method?

**Comment by JustAHuman-xD** [comment]
Or is it to prevent the block ticker from being added

**Comment by J3fftw1** [comment]
Im trying to prevent the block ticker from being added  + all the itemhandlers

**Comment by JustAHuman-xD** [comment]
Could we add some code to the register method? When it's state is set to disabled we remove its item handlers?

**Comment by JustAHuman-xD** [comment]
Tho looking at the code it looks like the BlockTicker is the only itemHandler that will actually really do anything. As loadItemHandlers is only called in onEnable, which is only called for items that are not disabled. While the BlockTicker is added to the registry in the addItemHandler method

**Comment by J3fftw1** [comment]
Andrew is gonna do the other PR.

**Review by JustAHuman-xD — DISMISSED** [review]
LGTM, could you test with Addons to make sure everything works properly?

**Comment by JustAHuman-xD** [comment]
Have you tested that latest commit?

**Comment by J3fftw1** [comment]
> Have you tested that latest commit?

Yes

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java [hunk 2]
```diff
@@ -475,10 +475,18 @@ public void register(@Nonnull SlimefunAddon addon) {
                 // The item has been disabled.
                 state = ItemState.DISABLED;
             }
-
+
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java:478]: 👀

**Review by JustAHuman-xD — COMMENTED** [review]
_this is the only thing_

**Review by JustAHuman-xD — APPROVED** [review]
LGTM 👍

**Review by Sefiraat — APPROVED** [review]
👍

**Pull request merged**